### DISCUSSION TITLE
Writing auth middleware

### DISCUSSION THREAD

**Discussion by mohammadjavad948** [tokio-rs/axum#1510] [Q&A] [answered]
Hi i want to write an auth middleware. there is an example in [docs](https://docs.rs/axum/latest/axum/middleware/fn.from_fn.html) but my question is how can i pass user info to route handler? for example my auth middleware looks at database and finds the user then i want to access user info on route handler. how can i do it?

**Comment by davidpdrsn** [comment, marked as answer]
Insert your data into a [request extension](https://docs.rs/http/latest/http/request/struct.Request.html#method.extensions_mut) and extract that in the handler.
